File: 프로그래머스/1/134240. 푸드 파이트 대회/푸드 파이트 대회.py
def solution(food):
    answer = ''
    
    for i in range(1,len(food)):
        for j in range(food[i]//2):
            answer += str(i)
            
    answer += '0'
    
    for i in range(len(food)-1, 0, -1):
        for j in range(food[i]//2):
            answer += str(i)
            
    return answer

File: 프로그래머스/1/134240. 푸드 파이트 대회/푸드 파이트 대회.java
class Solution {
    public String solution(int[] food) {
        String answer = "";
        StringBuilder sb = new StringBuilder();
        
        for(int i=1; i<food.length; i++) {
        	for(int j=0; j<food[i]/2; j++) {
        		sb.append(i);
        	}
        }
        
        String half = sb.toString();
        sb.append(0);
        for(int i=half.length()-1; i>-1; i--) {
        	sb.append(half.charAt(i));
        }
        
        answer = sb.toString();
        
        return answer;
    }
}